sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala

```diff
@@ -703,4 +703,53 @@ class DataStreamReaderWriterSuite extends StreamTest with BeforeAndAfter {
       queries.foreach(_.stop())
     }
   }
+
+  test("SPARK-32516: 'path' cannot coexist with load()'s path parameter") {
+    def verifyLoadFails(f: => DataFrame): Unit = {
+      val e = intercept[AnalysisException](f)
+      assert(e.getMessage.contains(
+        "Either remove the path option, or call load() without the parameter"))
+    }
+
+    verifyLoadFails(spark.readStream.option("path", "tmp1").parquet("tmp2"))
+    verifyLoadFails(spark.readStream.option("path", "tmp1").format("parquet").load("tmp2"))
+
+    withClue("SPARK-32516: legacy behavior") {
+      withSQLConf(SQLConf.LEGACY_PATH_OPTION_BEHAVIOR.key -> "true") {
+        spark.readStream
+          .format("org.apache.spark.sql.streaming.test")
+          .option("path", "tmp1")
+          .load("tmp2")
```

Comment:
This could be flaky. The directory of tmp2 could be non-empty and contains illegal data.

Reply:
This is using `org.apache.spark.sql.streaming.test` as a format, which has a no-op implementation: https://github.com/apache/spark/blob/f7995c576aa16f04681ca8f43f8c0f71818a9f44/sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala#L58-L59

This datasource is being used throughout this test suite with non-existent dirs: https://github.com/apache/spark/blob/f7995c576aa16f04681ca8f43f8c0f71818a9f44/sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala#L227-L231

Am I missing something? Thanks!

Reply:
You are right, I think this is not an issue here.